### DISCUSSION TITLE
No colors at all?

### DISCUSSION THREAD

**Discussion by telemachus** [nushell/nushell#10172] [Q&A] [answered]
Hi. I'm color blind, and I find all the colors distracting and hard to read (because hard to distinguish). I normally set my terminal (iTerm on macOS, if that matters) to black text on a white background.

With all that background, what is the simplest way to ask nushell not to color anything? To be clear, I'm not concerned about colors that non-built-in tools—say git—may add. I've already handled those. I want nushell itself not to color everything, and I'm hoping there is a global off button for the colors.)

Thanks in advance, and thanks for all the work on nushell.

**Comment by fdncred** [comment, marked as answer]
I think `use_ansi_coloring = false` in the config.nu should get you going. You can also try it out with `$env.config.use_ansi_coloring = false`. Please let us know if some colors leak out somewhere. Not many folks use this option.

**Reply by telemachus** [reply]
@fdncred Thanks. I tried `use_ansi_coloring = false` in config.nu and `$env.config.use_ansi_coloring = false` in env.nu, but neither seems to make any difference. The prompt and (e.g.) all the output of `ls` are still fully colored. I also tried removing the `color_config` setting from config.nu, but no joy. (Perhaps it defaults to some set of colors?)

One other note, I'm currently trying nushell out (from within a larger bash session). If I `exit` nushell to return to bash, everything is now colored in bash. I'm guessing this means that the color handling is missing a reset code for the terminal, but I can't be sure.

**Reply by fdncred** [reply]
> $env.config.use_ansi_coloring = false in env.nu, but neither seems to make any difference.

I just did this `$env.config.use_ansi_coloring = false` and did a `ls` and it's totally stripped of colors, so it does something.

The prompt calls are handled separately and apparently do not respect this config point. That's a good case for a PR.

Maybe you're on an old version or something? The latest released version is 0.84.0.

**Reply by telemachus** [reply]
> I just did this $env.config.use_ansi_coloring = false and did a ls and it's totally stripped of colors, so it does something.

I see now that I misunderstood you. I tried adding `$env.config.use_ansi_coloring = false` to the top (and bottom) of my env.nu file, and that doesn't work. I will need to figure out the syntax of those files, clearly. If I run it in an open session, then—yes—colors stop. Thanks. I'll need to investigate more to see how to make that configuration option permanent and whether something else in my shell configuration (under bash) is affecting things. (I use a [ridiculously complicated PROMPT_COMMAND](https://gist.github.com/telemachus/85ac94cc89054a3335fa447038042d58) to control my  prompt and iTerm's titlebar at the same time. I think that may be causing problems here.)

**UPDATE**: Yup, it was my fault all along. `use_ansi_coloring: false` in config.nu does the job if I kill my PROMPT_COMMAND (e.g., `export PROMPT_COMMAND= ; nu`). I apologize for the trouble.

**Reply by fdncred** [reply]
No worries at all. `$env.config.use_ansi_coloring` is short-hand for "edit your config.nu, find the `$env.config` variable, now search for `use_ansi_coloring` and set it to `false`". I should've been more clear.

Glad it's working for you now. Let us know if colors creep in somewhere.

**Comment by amtoine** [comment]
@telemachus
maybe we could have a `color-blind.nu` theme in the `nu_scripts` for that scenario?

i'm sure you're not the only one having that kind of accessibility issues :open_mouth: 

because i do not have issues with colors, i can not think of anything different than a pure-white theme, which is not necessarily the best :sweat_smile: 
maybe we'll need some help :innocent:

**Reply by telemachus** [reply]
> maybe we could have a `color-blind.nu` theme in the `nu_scripts` for that scenario?
> 
> i'm sure you're not the only one having that kind of accessibility issues 😮
> 
> because i do not have issues with colors, i can not think of anything different than a pure-white theme, which is not necessarily the best 😅 maybe we'll need some help 😇

I'm willing to help, but I'll need to learn how the color configuration works. I was hoping to avoid configuration since what I want is (literally) black text on a white background, but hopefully it won't be hard to write such a theme. I'll look into it over the next few days.

**Reply by amtoine** [reply]
no hurry :relieved: 
and as `$env.config` can be changed to remove the colors, that might be just enough :yum:
